pygeohash/geohash.py

```diff
@@ -19,20 +19,33 @@
 <http://www.gnu.org/licenses/>.
 
 Modified 2015 by Will McGinnis for pygeohash.
+Modified 2023 by Paarth Shah for pygeohash.
 """
+from __future__ import annotations
 
 from math import log10
+from typing import Dict, Tuple, NamedTuple
 
 #  Note: the alphabet in geohash differs from the common base32 alphabet described in IETF's RFC 4648
 # (http://tools.ietf.org/html/rfc4648)
 
 __base32 = '0123456789bcdefghjkmnpqrstuvwxyz'
-__decodemap = dict()
-for i in range(len(__base32)):
-    __decodemap[__base32[i]] = i
+__decodemap: Dict[str, int] = {base32_char: i for i, base32_char in enumerate(__base32)}
 
 
-def decode_exactly(geohash):
+class LatLong(NamedTuple):
+    latitude: float
+    longitude: float
+
+
+class ExactLatLong(NamedTuple):
+    latitude: float
+    longitude: float
+    latitude_error: float
+    longitude_error: float
```

Comment:
Fully compatible with a normal `tuple` for all downstream purposes, but exposes named attribute access to `.latitude` and `.longitude` etc as well as indexed access